gflownet/utils/buffer.py

```diff
@@ -259,11 +259,47 @@ def make_data_set(self, config):
         )
         return df, {"x": samples, "energy": energies}
 
-    def compute_stats(self, data):
+    @staticmethod
+    def compute_stats(data):
         mean_data = data["energies"].mean()
         std_data = data["energies"].std()
         min_data = data["energies"].min()
         max_data = data["energies"].max()
         data_zscores = (data["energies"] - mean_data) / std_data
         max_norm_data = data_zscores.max()
         return mean_data, std_data, min_data, max_data, max_norm_data
+
+    @staticmethod
+    def select(data_dict: dict, n: int, mode: str = "permutation", rng=None):
+        if n == 0:
+            return []
+        samples = data_dict["x"]
+        # If the data_dict comes from the replay buffer, then samples is a dict and we
+        # need to keep its values only
+        if isinstance(samples, dict):
+            samples = list(samples.values())
+        if mode == "permutation":
+            assert rng is not None
+            samples = [samples[idx] for idx in rng.permutation(n)]
+        elif mode == "weighted":
+            if "rewards" in data_dict:
+                score = "rewards"
+            elif "energy" in data_dict:
+                score = "energy"
```

Comment:
Is there no scenario in which you want `weighted` and `energy` but the dict contains `rewards` and so it never gets to `score = "energy"`?

Reply:
Oh the thing is it will be one thing or the other (`energy` or `rewards`) in the dictionary, depending on whether it's coming from a replay buffer (`rewards`) or a data set (`energy`). And if it happened to have both, it's fine to use the rewards if they are available.

Reply:
SGTM